libs/utils/sidekick-decorate.js

```diff
@@ -1,9 +1,124 @@
 import userCanPublishPage from '../tools/utils/publish.js';
+import { hasPreflightFailures, executePreflightChecks } from '../blocks/preflight/checks/preflightExecutor.js';
+import { loadStyle } from './utils.js';
 
 const PUBLISH_BTN = '.publish.plugin button';
 const PROFILE = '.profile-email';
 const CONFIRM_MESSAGE = 'Are you sure? This will publish to production.';
 let stylePublishCalled = false;
+let preflightNotificationDismissed = false;
+
+function openPreflightPanel() {
+  const sk = document.querySelector('aem-sidekick, helix-sidekick');
+  if (sk) {
+    sk.dispatchEvent(new CustomEvent('custom:preflight', { bubbles: true }));
+  }
+}
+
+async function createPreflightNotification() {
+  const existingNotification = document.querySelector('.milo-preflight-overlay');
+  if (existingNotification) {
+    existingNotification.remove();
+  }
+  loadStyle('/libs/utils/preflight-notification.css');
+
+  const overlay = document.createElement('div');
+  overlay.className = 'milo-preflight-overlay';
+  overlay.innerHTML = `
+    <div class="preflight-notification">
+      <div class="notification-content">
+        <span class="notification-message">
+          Content quality checks are failing. Please <button class="preflight-review-link">review</button> before publishing.
+        </span>
+        <button class="notification-close">×</button>
+      </div>
+    </div>
+  `;
+
+  overlay.style.cssText = `
+    position: fixed;
+    bottom: 120px;
+    left: 50%;
+    transform: translateX(-50%);
+    z-index: 10000;
+  `;
+
+  const reviewLink = overlay.querySelector('.preflight-review-link');
+  reviewLink.addEventListener('click', (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    openPreflightPanel();
+    overlay.remove();
+  });
+
+  const closeBtn = overlay.querySelector('.notification-close');
+  closeBtn.addEventListener('click', () => {
+    overlay.remove();
+    preflightNotificationDismissed = true;
+  });
+
+  document.body.appendChild(overlay);
+}
+
+async function checkPreflightAndShowNotification() {
+  try {
+    await executePreflightChecks();
+    const hasFailures = hasPreflightFailures();
+    const existingNotification = document.querySelector('.milo-preflight-overlay');
+    if (existingNotification) {
+      existingNotification.remove();
+    }
+
+    if (hasFailures && !preflightNotificationDismissed) {
+      await createPreflightNotification();
+    }
+  } catch (error) {
+    console.warn('Failed to check preflight status:', error);
+  }
+}
+
+function createSidekickVisibilityObserver() {
+  const observer = new MutationObserver(async () => {
+    const sidekick = document.querySelector('aem-sidekick');
```

Comment:
Suggestion: create a method that returns the sidekick (if available) and reuse it:
* https://github.com/adobecom/milo/pull/4496/files#diff-8fca44a004d5ac9b5fd40efac647ac0deb02ada1314e83f34593176290dcbb53R12
* https://github.com/adobecom/milo/pull/4496/files#diff-8fca44a004d5ac9b5fd40efac647ac0deb02ada1314e83f34593176290dcbb53R82
* https://github.com/adobecom/milo/pull/4496/files#r2180167699